consensus/src/chained_bft/sync_manager.rs

```diff
@@ -211,24 +211,8 @@ where
         self.storage
             .save_tree(blocks.clone(), quorum_certs.clone())?;
         let pre_sync_instance = Instant::now();
-        match self
-            .state_computer
-            .sync_to(highest_ledger_info.clone())
-            .await
-        {
-            Ok(true) => (),
-            Ok(false) => panic!(
-                "state synchronizer failure, this validator will be killed as it can not \
-                 recover from this error.  After the validator is restarted, synchronization will \
-                 be retried.",
-            ),
-            Err(e) => panic!(
-                "state synchronizer failure: {:?}, this validator will be killed as it can not \
-                 recover from this error.  After the validator is restarted, synchronization will \
-                 be retried.",
-                e
-            ),
-        };
+        self.state_computer
+            .sync_to_or_bail(highest_ledger_info.clone());
```

Comment:
This code changes the async wait to the synchronous one, but it's ok because the thread that is running the state synchronization logic doesn't have any other tasks in the current design.